lib/TH/generic/THStorage.h

```diff
@@ -28,6 +29,8 @@ typedef struct THStorage
     char flag;
     THAllocator *allocator;
     void *allocatorContext;
+    int type;
```

Comment:
Sorry, just saw the comment in the description. What I don't like about it is the `TH_Cuda` type. That should really be in cutorch, shouldn't it?